pkgs/development/python-modules/smbus-cffi/default.nix

```
@@ -0,0 +1,46 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, cffi
+, pytestCheckHook
+, pyserial
+, fetchpatch
+}:
+
+buildPythonPackage rec {
+  pname = "smbus-cffi";
+  version = "0.5.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "fb4195aaabfc01586863f60d3190b5cb1bf8f12622fd597e23e48768dad6bde8";
+  };
+
+  patches = [
+    # https://github.com/bivab/smbus-cffi/pull/24
+    (fetchpatch {
+      url = "https://github.com/bivab/smbus-cffi/commit/ba79ae174a9d84e767d95f165c43ee212b1bbb92.patch";
+      sha256 = "sha256-WtRuK5y6fWDEhm0Xy5XqS5yCkn7vXXYtjlOjS90gla4=";
+    })
+  ];
+
+  propagatedBuildInputs = [ cffi ];
+
+  installCheckPhase = ''
+    # we want to import the installed module that also contains the compiled library
+    rm -rf smbus
+    runHook pytestCheckPhase
+  '';
+
+  # requires hardware access
+  pytestFlagsArray = [ "--ignore=test/test_smbus_integration.py" ];
+
+  checkInputs = [ pytestCheckHook pyserial ];
+
+  meta = with lib; {
+    description = "This Python module allows SMBus access through the I2C /dev interface on Linux hosts. The host kernel must have I2C support, I2C device interface support, and a bus adapter driver";
```

This description is probably too long and detailed?

```suggestion
    description = "Python module for SMBus access through Linux I2C /dev interface";
```